apps/beeswax/src/beeswax/server/dbms.py

```diff
@@ -63,18 +63,40 @@
 
 LOG = logging.getLogger(__name__)
 
+
+RESET_HS2_QUERY_SERVER = False
 DBMS_CACHE = {}
 DBMS_CACHE_LOCK = threading.Lock()
 cache = caches[CACHES_HIVE_DISCOVERY_KEY]
 
 # Using file cache to make sure eventlet threads are uniform, this cache is persistent on startup
 # So we clear it to make sure the server resets hiveserver2 host.
-cache.clear()
+def reset_ha():
+  global RESET_HS2_QUERY_SERVER
+  cache.clear()
+  cache.clear()
+  RESET_HS2_QUERY_SERVER = True
+
+
+reset_ha()
+
+
+def get_zk_hs2():
+  hiveservers = None
+  zk = KazooClient(hosts=libzookeeper_conf.ENSEMBLE.get(), read_only=True)
+  zk.start()
+  znode = HIVE_DISCOVERY_HIVESERVER2_ZNODE.get()
+  if zk.exists(znode):
+    LOG.debug("Selecting up Hive server via the following node {0}".format(znode))
+    hiveservers = zk.get_children(znode)
+  zk.stop()
+  return hiveservers
 
 
 def get(user, query_server=None, cluster=None):
   global DBMS_CACHE
   global DBMS_CACHE_LOCK
+  global RESET_HS2_QUERY_SERVER
```

Comment:
We declared this variable global in line 75, is it required to be done again?

Reply:
@Harshg999 , yes this was needed else it fails with RESET variable issue